polkadot/node/core/candidate-validation/src/lib.rs

```diff
@@ -1233,3 +1271,16 @@ fn pvf_exec_timeout(executor_params: &ExecutorParams, kind: PvfExecKind) -> Dura
 		PvfExecKind::Approval => DEFAULT_APPROVAL_EXECUTION_TIMEOUT,
 	}
 }
+
+fn pvf_exec_deadline(
+	exec_timeout: Duration,
+	validation_request_ttl: Option<Duration>,
+) -> Option<Instant> {
+	validation_request_ttl.and_then(|ttl| {
+		if ttl <= exec_timeout {
```

Comment:
What's the rationale here ?

Reply:
It takes place in tests where parameters are set randomly.